scripts-setup/setup-pre-commit-hook.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/sh
+set -e
+GITDIR=$(git rev-parse --show-toplevel)
+GITHOOK=${GITDIR}/.git/hooks/prepare-commit-msg 
+if [ -e ${GITHOOK} ]; then
+    echo "A prepare commit message script already exists at ${GITHOOK}"
+    echo "please delete it if you want to replace it.  Alternatively, you can"
+    echo "merge the logic in this script with the existing logic in"
+    echo "${GITHOOK}"
+    exit 1
+fi
+cat << 'EOF' > ${GITHOOK}
+#/bin/sh
+NAME=$(git config user.name)
+EMAIL=$(git config user.email)
+if [ -z "$NAME" ]; then
+    echo "empty git config user.name"
+    exit 1
+fi
+if [ -z "$EMAIL" ]; then
+    echo "empty git config user.email"
+    exit 1
+fi
+git interpret-trailers --if-exists doNothing --trailer \
+    "Signed-off-by: $NAME <$EMAIL>" \
+     --in-place "$1" 
+EOF
```

Comment:
All of the occurrences of `$` need to be escaped in the script emitted via the `cat`.

Maybe it would be better to create directory to hold this script and any other git hooks we might want to provide, then have the setup script copy or link those directly?

Reply:
> All of the occurrences of $ need to be escaped in the script emitted via the cat.

Does [this trick](https://stackoverflow.com/a/22698106/1446624) not work in some scenario I don't know about?

> Maybe it would be better to create directory to hold this script and any other git hooks we might want to provide, then have the setup script copy or link those directly?

Sure, that's probably more clear.  I could do a `git-scripts` subdirectory and rename this script `setup-git-hooks.sh`

Reply:
> Does this trick not work in some scenario I don't know about?

Well, yes it does... that's a new one on me!

>  I could do a git-scripts subdirectory and rename this script setup-git-hooks.sh

Sounds good, thanks!

Reply:
Done!  I also modified to setup submodules properly after I realized the script was not working on `meta-tegra` submodule commits.  Let me know if you'd like any further changes.